django_webapps/settings_sample.py

```diff
@@ -10,6 +10,10 @@
 For the full list of settings and their values, see
 https://docs.djangoproject.com/en/4.1/ref/settings/
 """
+# This import is for mypy pre-commit hook. Otherwise, list[str] in ALLOWED_HOSTS triggers an error
+# despite Python version being specified in the args. See:
+# https://mypy.readthedocs.io/en/stable/runtime_troubles.html#future-annotations
```

Comment:
Are you sure about it? I've just tested, for me it works fine without this import

Reply:
The funny thing is that when I run `pre-commit run -a`, it works. But when I commit in PyCharm with the hooks enabled, it fails!

Reply:
You know what, I think the reason may be that when I try to commit. the pre-commit hooks run in the system-wide Python version, not in my venv (that's why you have to list all the dependencies, right?). And the system-wide Python version on Ubuntu 20.04 is 3.8. I have 3.10 installed too, obviously, but when I type `python3`, I get Python 3.8.

Reply:
But I'm still thinking... maybe I should just do `from typing import List`

Reply:
AFAIK pre-commit makes it's own venv(s), so it shouldn't break. But I'm really not sure where is trouble at your side :(

Reply:
Maybe try to do 
`pre-commit clean`
`pre-commit uninstall`
`pre-commit install`
`pre-commit run`

Reply:
Okay that worked. Really weird.